index.js

```diff
@@ -0,0 +1,59 @@
+const core = require('@actions/core');
+// const github = require('@actions/github');
+
+const { workspaces } = require('./package.json');
+const env = 'staging';
+
+try {
+    // `files` input defined in action metadata file
+    const input = core.getInput('files');
+    // We're receiving this stringified, parse it in order to use it as an array
+    const changedFiles = Array.isArray(input) ? input : JSON.parse(input);
+    console.log(`changed files: ${changedFiles}`);
+    const inWorkspace = [];
+    changedFiles.forEach((file) => {
+        let integration;
+        if (file.includes('/')) {
+            integration = file.split('/')[0];
+        } else if (file.includes('\\')) {
+            /* Should support Backslash as well as forward ?*/
+            file.split('\\')[0];
+        } else {
+            // TODO could address root folders files here such as package.json (readme, .gitignore, .env, That's it ? )
+            // Shouldn't
+            // integration = file;
+        }
+
+        if (
+            workspaces.includes(integration) &&
+            !inWorkspace.includes(integration)
+        ) {
+            inWorkspace.push(integration);
+        } else {
+            // Could do with else if that checks custom changes that are ok before throwing ?
+            // for example: package.json, .github/workflows
+            if (inWorkspace.includes(integration)) {
+                console.log(
+                    `Already added ${integration} to integrations that will be released`
+                );
+            }
+            console.log(`${file} is not in any of the workspaces`);
+            // The above log should be:
```

Comment:
not sure we should 'throw' in 'while' we will mis the other correct files. or do you want to fail it all?

Reply:
Good question, I thought about it a bit but would like to hear your opinion as well.
I think failing it is the best approach since otherwise there could be a scenario where half of the integrations are deployed and half aren't and the developer doesn't receive any indication and just carries on with his life.
Or we could not fail it and just output a message saying which integrations releases failed.